### PR TITLE
[_163] Allow custom irods username claim

### PR DESCRIPTION
**Pull request by MartinFlores751**
Provides a configurable claim name for the mapping of OIDC users to iRODS users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -268,6 +268,10 @@ Notice how some of the configuration values are wrapped in angle brackets (e.g.
                 // The amount of time before the OIDC Authorization Code grant
                 // times out, requiring another attempt at authentication.
                 "state_timeout_in_seconds": 600
+
+                // Name to claim search for to provide mapping of OIDC user
```
Comment (trel) [inline README.md:272]: not sure what this means... "Name to claim search for"

also, how does this relate to our possible future of having alternate mappings?

Comment (MartinFlores751) [reply README.md:272]: Whoops, issue between thoughts and typing it seems.

I believe this option would only relate to the current user mapping with the OIDC claims. As for 'is this optional' or 'how would this interact if we use mapping B or C', not sure how that would work out.

File: README.md [hunk 2]
```diff
@@ -268,6 +268,10 @@ Notice how some of the configuration values are wrapped in angle brackets (e.g.
                 // The amount of time before the OIDC Authorization Code grant
                 // times out, requiring another attempt at authentication.
                 "state_timeout_in_seconds": 600
+
+                // The name of the OIDC claim which provides the mapping of a
+                // OIDC user to a irods user account
```
Comment (korydraughn) [inline README.md:273]: "a irods" -> "an iRODS"

**Review by alanking — COMMENTED** [review]
Nice

File: README.md (hunk 2 above)
Comment (trel) [reply README.md:273]: same with OIDC i think, if we're pronouncing it oh-eye-dee-see.

"a OIDC" -> "an OIDC"

Comment (trel) [reply README.md:273]: oh, reading from bottom to top... i duplicated... carry on...

File: README.md [hunk 3]
```diff
@@ -268,6 +268,10 @@ Notice how some of the configuration values are wrapped in angle brackets (e.g.
                 // The amount of time before the OIDC Authorization Code grant
                 // times out, requiring another attempt at authentication.
                 "state_timeout_in_seconds": 600
+
+                // The name of the OIDC claim which provides the mapping of a
```
[the remark this answers is not in the document]
Comment (MartinFlores751) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (MartinFlores751) [reply README.md]: Done.

**Comment by MartinFlores751** [comment]
All tests pass, save for one:
`test_server_does_not_crash_when_incorrect_http_method_is_used_to_authenticate - AssertionError: 200 != 405`

**Comment by korydraughn** [comment]
Please run clang-format. Investigate why the test fails. I'll look at well.

Is there overlap between this PR and the other one.

**Comment by korydraughn** [comment]
All tests pass on tip of main.

Seems there's some difference between our setups that the tests don't handle yet.

**Comment by korydraughn** [comment]
There indeed real test failures. I was testing against unreleased iRODS server code. There are 3 failing tests and non of them are related to your work.

I'll open an issue for the failures and get them resolved.

Please squash to taste and leave off the pounds.

**Comment by MartinFlores751** [comment]
Squashed.

**Review by korydraughn — APPROVED** [review]
Pound it!

**Comment by korydraughn** [comment]
Oh, please change `irods` to `iRODS` in the commit message.

**Comment by MartinFlores751** [comment]
Rebased, running tests.

**Comment by MartinFlores751** [comment]
Got one failure:
`test_server_does_not_crash_when_incorrect_http_method_is_used_to_authenticate - AssertionError: 200 != 405`

**Comment by korydraughn** [comment]
Which line/assertion failed?

**Comment by korydraughn** [comment]
Does your config define OIDC options?

**Comment by MartinFlores751** [comment]
Yes, both this and #197 have OIDC defined. The other PR changes the `oidc` to `openid_connect` though. Don't think it makes a difference for this test, but mentioning jic.

**Comment by korydraughn** [comment]
Try hiding the openid_connect (i.e. add a leading underscore or something) config and running the tests again.

That test assumes the openid_connect block doesn't exist.

**Comment by korydraughn** [comment]
Excellent. If the OIDC tests pass, update the commit message to use iRODS instead of irods and let's get this merged.

**Comment by MartinFlores751** [comment]
Reworded & pounded

**Review by alanking — APPROVED** [review]

**Pull request merged**